File: v2/sdk/javascript/examples/with-profile.ts
/**
 * Prediction with full country/budget/timeline context.
 * Run: npx tsx examples/with-profile.ts
 */
import { SimulatorClient, RateLimitError, ValidationError } from '../index';

async function main() {
  const client = new SimulatorClient({
    baseUrl: 'https://simulator.vercel.app',
    timeout: 60000, // longer timeout for complex scenarios
  });

  try {
    const result = await client.predict({
      scenario: 'Build a fintech startup offering micro-loans to small businesses',
      country: 'Nigeria',
      budget: 250000,
      timeline: '2 years',
    });

    console.log('=== PREDICTION ===');
    console.log(`Scenario: ${result.scenario}`);
    console.log(`Overall probability: ${result.probability}%`);
    console.log(`Confidence in data: ${(result.confidence * 100).toFixed(0)}%`);
    console.log(`Best case: ${result.probRange.optimistic}%`);
    console.log(`Worst case: ${result.probRange.adverse}%`);
    console.log('');

    console.log('=== BOTTLENECKS ===');
    for (const b of result.keyBottlenecks) {
      const bar = '#'.repeat(Math.round(b.prob / 5));
      console.log(`  ${b.label.padEnd(25)} [${String(b.prob).padStart(3)}%] ${bar}`);
      console.log(`  ${''.padEnd(25)} ${b.description}`);
    }
    console.log('');

    console.log('=== DATA SOURCES ===');
    for (const s of result.sources) {
      console.log(`  - ${s}`);
    }
    console.log('');

    console.log('=== SACRED ROOTS ===');
    for (const r of result.sacredRoots) {
      console.log(`  - ${r}`);
    }

    if (result._meta) {
      console.log(`\n[Provider: ${result._meta.provider}, Data: ${result._meta.dataSource}]`);
    }
  } catch (err) {
    if (err instanceof ValidationError) {
      console.error('Validation failed:', err.message);
    } else if (err instanceof RateLimitError) {
      console.error(`Rate limited. Reset at: ${err.resetAt.toISOString()}`);
    } else {
      console.error('Error:', (err as Error).message);
    }
  }
}

main();

File: v2/sdk/python/simulator_sdk/examples/with_profile.py
"""
Prediction with full country/budget/timeline context.
Run: python -m simulator_sdk.examples.with_profile
"""
from simulator_sdk import SimulatorClient, RateLimitError, ValidationError


def main():
    client = SimulatorClient(
        base_url="https://simulator.vercel.app",
        timeout=60,  # longer timeout for complex scenarios
    )

    try:
        result = client.predict(
            scenario="Build a fintech startup offering micro-loans to small businesses",
            country="Nigeria",
            budget=250000,
            timeline="2 years",
        )

        print("=== PREDICTION ===")
        print(f"Scenario: {result.scenario}")
        print(f"Overall probability: {result.probability}%")
        print(f"Confidence in data: {result.confidence * 100:.0f}%")
        print(f"Best case: {result.prob_range['optimistic']}%")
        print(f"Worst case: {result.prob_range['adverse']}%")
        print()

        print("=== BOTTLENECKS ===")
        for b in result.key_bottlenecks:
            bar = "#" * round(b.prob / 5)
            print(f"  {b.label:<25} [{b.prob:>3}%] {bar}")
            print(f"  {'':25} {b.description}")
        print()

        print("=== DATA SOURCES ===")
        for s in result.sources:
            print(f"  - {s}")
        print()

        print("=== SACRED ROOTS ===")
        for r in result.sacred_roots:
            print(f"  - {r}")

        if result.meta:
            print(f"\n[Provider: {result.meta.get('provider')}, Data: {result.meta.get('dataSource')}]")

    except ValidationError as e:
        print(f"Validation failed: {e}")
    except RateLimitError as e:
        print(f"Rate limited. Reset at: {e.reset_at.isoformat()}")
    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    main()
